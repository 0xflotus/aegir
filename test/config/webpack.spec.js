/* eslint-env jest */
'use strict'

const sinon = require('sinon')
const path = require('path')

describe('config - webpack', () => {
  let config
  beforeEach(() => {
    sinon.stub(process, 'cwd').returns(path.join(__dirname, 'fixtures'))
    config = require('../../src/config/webpack')
  })

  afterEach(() => {
    process.cwd.restore()
  })

  it('custom configs', () => {
    expect(config).toHaveProperty('output.filename', 'main.js')
    expect(config).toHaveProperty('output.library', 'Example')
    expect(config).toHaveProperty('devtool', 'eval')

    expect(config).toMatchSnapshot()
  })
})
