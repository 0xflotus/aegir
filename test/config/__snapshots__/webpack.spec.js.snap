// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config - webpack custom configs 1`] = `
Object {
  "devtool": "eval",
  "entry": Array [
    "/Users/dignifiedquire/opensource/ipfs/aegir/src/main.js",
  ],
  "module": Object {
    "rules": Array [
      Object {
        "loader": "json-loader",
        "test": /\\\\\\.json\\$/,
      },
    ],
  },
  "node": Object {
    "Buffer": true,
  },
  "output": Object {
    "filename": "main.js",
    "library": "Example",
    "path": "/Users/dignifiedquire/opensource/ipfs/aegir/test/config/fixtures/dist",
  },
  "performance": Object {
    "hints": false,
  },
  "plugins": Array [],
  "resolve": Object {
    "alias": Object {
      "http": "stream-http",
      "zlib": "browserify-zlib",
    },
    "modules": Array [
      "node_modules",
      "/Users/dignifiedquire/opensource/ipfs/aegir/src/node_modules",
    ],
  },
  "resolveLoader": Object {
    "moduleExtensions": Array [
      "-loader",
    ],
    "modules": Array [
      "node_modules",
      "/Users/dignifiedquire/opensource/ipfs/aegir/src/node_modules",
    ],
  },
  "target": "web",
}
`;
